Implement Guzzle v6

Echron opened the pull request:
Implementation of latest Guzzle client
+ Add reporting example

arendjantetteroo commented:
Thanks! Will try to look into this soon.

Is it backwards compatible? So for users of this client only the dependencies change? Or do we need a new major version so we dont break current implementations?

arendjantetteroo commented on src/AJT/Toggl/ReportsClient.php:
```diff
@@ -23,38 +20,25 @@ class ReportsClient extends Client
-            "Content-type" => "application/json",
-        ));
+        if (isset($config['apiVersion']) && $config['apiVersion'] !== 'v2') {
+            throw new \Exception('Only v8 is supported at this time');
```
Only v2 for reporting is supported at this time?

Echron replied:
Ah yes, copy paste error. I'm not sure why to keep 2 clients who do almost the same except for the different definition file. But i'm not gonna change that ;)

arendjantetteroo commented:
Thanks @Echron , tried it out, the examples seem to work great still. 

Will add a new major version

Echron commented:
I refactored it to be backwards compatible, i wrote some simple tests which all worked after the refactoring. But since there are no unit tests i'm not 100% sure. I tested with all the existing examples and added one for reporting.

arendjantetteroo commented:
Released 1.0 with your work. Thanks again.
